### PR TITLE
[GPU] Fix warnings

### PR DESCRIPTION
**Pull request by andrei-cv**
### Details:
 - *fix warning in GPU plugin and GPU tests*

### Tickets:
 - *93716*

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp [hunk 1]
```diff
@@ -827,7 +827,7 @@ static bool is_node_for_onednn(fully_connected_node const& node) {
     auto fc_prim = node.get_primitive();
     auto ps = node.get_output_layout().get_partial_shape();
     int non_spatial_count = 2 + (fc_prim->input_size == 3 ? 1 : 0);
-    int rank = ps.size();
+    int rank = static_cast<int>(ps.size());
```
Comment (ilya-lavrenov) [inline src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:830]: maybe we need to change `non_spatial_count `to be `size_t` instead?

File: src/plugins/intel_gpu/src/graph/network.cpp [hunk 2]
```diff
@@ -541,7 +541,7 @@ void network::save(cldnn::BinaryOutputBuffer& ob) {
     }
 
     int exec_order_size;
-    exec_order_size = _exec_order.size();
+    exec_order_size = static_cast<int>(_exec_order.size());
     ob << exec_order_size;
```
Comment (ilya-lavrenov) [inline src/plugins/intel_gpu/src/graph/network.cpp:545]:
```c++
ob << _exec_order.size();
```
?

File: src/plugins/intel_gpu/src/kernel_selector/kernels/scatter_update/scatter_nd_update_kernel_ref.cpp [hunk 3]
```diff
@@ -130,7 +130,7 @@ static std::string GetInputBlockND(const scatter_nd_update_params& params, size_
     block_nd_s[rank] = "1";
     size_t input_offset = num * 6;
 
-    for (int32_t idx = (rank - 1); idx >= 0; --idx) {
+    for (int32_t idx = static_cast<int32_t>(rank - 1); idx >= 0; --idx) {
```
Comment (ilya-lavrenov) [inline src/plugins/intel_gpu/src/kernel_selector/kernels/scatter_update/scatter_nd_update_kernel_ref.cpp:133]: idx -> size_t ?

File: src/plugins/intel_gpu/tests/test_cases/deconvolution_gpu_test.cpp [hunk 4]
```diff
@@ -2425,9 +2425,9 @@ struct deconvolution_random_test_params {
         };
 
         auto print_strides = [&](const ov::Strides& s) {
-            std::string res = to_string_neg(s[0]);
+            std::string res = to_string_neg(static_cast<int>(s[0]));
```
Comment (ilya-lavrenov) [inline src/plugins/intel_gpu/tests/test_cases/deconvolution_gpu_test.cpp:2428]: do we really need to change type before converting to string?

File: src/plugins/intel_gpu/tests/test_cases/lru_caches_gpu_test.cpp [hunk 5]
```diff
@@ -43,11 +43,11 @@ TEST(lru_cache, basic_data_type)
 
     std::vector<std::pair<int, int>> expected_value;
     for (size_t i = ca.size(); i > 0; i--) {  // 5, 1, 2, 4
-        int idx = input_values.size() - i;
+        int idx = static_cast<int>(input_values.size() - i);
```
Comment (ilya-lavrenov) [inline src/plugins/intel_gpu/tests/test_cases/lru_caches_gpu_test.cpp:46]: idx -> size_t ?

**Review by ilya-lavrenov — COMMENTED** [review]
In some places it looks like we need to change the types instead of simply adding static_casts

**Review by sshlyapn — APPROVED** [review]

**Pull request merged**